### PR TITLE
/app/pages/common/Modals/PreferencesModal/Badges/index.js 🔨 Refactor 🧠 Overmind Hacktoberfest

### PR DESCRIPTION
**Pull request by Sowmiya07**
Refactored the badges index.js file to use overmind and added named export instead of default export

## What kind of change does this PR introduce?
This is part of the requested refactor for hacktoberfest mentioned in #2760
@Saeris @christianalfoni

## What is the current behavior?
Previously it was utilizing inject and hooksObserver from app/componentConnectors.

## What is the new behavior?
Prefers useOvermind over inject and other small stylistic choices mentioned in #2760

## What steps did you take to test this? This is required before we can merge.
yarn:lint: I was not able to do this as  I have some errors doing yarn
yarn:test: I was not able to do this as  I have some errors doing yarn

## Checklist

- [ ] Documentation
- [ ] Testing 
- [ ] Ready to be merged
      
- [ ] Added myself to contributors table

### CODE REVIEW

**Comment by ganes1410** [comment]
@Sowmiya07 I think this is failing because of lint error. Maybe try running `yarn lint` and try to push again

File: packages/app/src/app/pages/common/Modals/PreferencesModal/Badges/index.tsx [hunk 1]
```diff
@@ -1,12 +1,20 @@
-import React from 'react';
-import { inject, observer } from 'app/componentConnectors';
+import React, {FunctionComponent} from 'react';
```
Comment (Saeris) [inline packages/app/src/app/pages/common/Modals/PreferencesModal/Badges/index.tsx]:
```suggestion
import React from 'react';
```

For types we don't need to use named imports, we can just pull it off React

File: packages/app/src/app/pages/common/Modals/PreferencesModal/Badges/index.tsx [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+import { useOvermind } from 'app/overmind';
 
 import Margin from '@codesandbox/common/lib/components/spacing/Margin';
 import Badge from '@codesandbox/common/lib/utils/badges/Badge';
 import { Title } from '../elements';
 
-function BadgesComponent({ store, signals }) {
-  const badgesCount = store.user.badges.length;
+export const Badges: FunctionComponent = () => {
```
Comment (Saeris) [inline packages/app/src/app/pages/common/Modals/PreferencesModal/Badges/index.tsx]:
```suggestion
export const Badges: React.FC = () => {
```

File: packages/app/src/app/pages/common/Modals/PreferencesModal/Badges/index.tsx [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+  const {
+    state: {
+      user: {badges},
```
Comment (Saeris) [inline packages/app/src/app/pages/common/Modals/PreferencesModal/Badges/index.tsx]:
```suggestion
      user: { badges },
```

eslint --fix should have caught this, not sure why it didn't. Adding spaces for consistency

File: packages/app/src/app/pages/common/Modals/PreferencesModal/Badges/index.tsx [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
+    },
+    actions: {
+      preferences: {setBadgeVisibility},
```
Comment (Saeris) [inline packages/app/src/app/pages/common/Modals/PreferencesModal/Badges/index.tsx]:
```suggestion
      preferences: { setBadgeVisibility },
```

**Comment by SaraVieira** [comment]
Thank you for your help!

**Pull request merged**